contracts/core/extensions/CoreIssuanceOrder.sol

```diff
@@ -272,4 +275,57 @@ contract CoreIssuanceOrder is
             INVALID_QUANTITY
         );
     }
+
+    function settleOrder(
+        OrderLibrary.IssuanceOrder _order,
+        uint _fillQuantity,
+        bytes _orderData
+    )
+        private
+    {
+        uint[] requiredBalances;
+
+        // Calculate amount of maker token required
+        // Look into rounding errors
+        uint requiredMakerToken = _fillQuantity.mul(_order.makerTokenAmount).div(_order.quantity);
+
+        // Get Set Token information
+        address[] memory components = ISetToken(_order.setAddress).getComponents();
+        uint[] memory componentUnits = ISetToken(_order.setAddress).getUnits();
+
+        // Calculate amount of component tokens required to issue
+        for (uint16 i = 0; i < components.length; i++) {
+            // Get current vault balances
+            uint tokenBalance = IVault(state.vaultAddress).getOwnerBalance(
+                _order.makerAddress,
+                components[i]
+            );
+
+            // Calculate amount needed to issue _fillQuantity amount
+            uint requiredAddition = componentUnits[i].mul(_fillQuantity);
+
+            // Required vault balances after exchange order executed
+            requiredBalances.push(tokenBalance.add(requiredAddition));
+        }
+
+        // Execute exchange orders
+        uint makerTokenUsed = executeExchangeOrders(_orderData);
+        require(makerTokenUsed <= requiredMakerToken);
+
+        // Check that maker's component tokens in Vault have been incremented correctly
+        for (uint16 i = 0; i < components.length; i++) {
+            uint currentBal = IVault(state.vaultAddress).getOwnerBalance(
```

Comment:
Thinking through the possibilities and vulns here - `issue` transfers from the user account if the required component amount is not in the vault.